Cabal/Distribution/Simple/GHC.hs

```diff
@@ -297,6 +298,18 @@ getInstalledPackages verbosity comp packagedbs conf = do
         _  -> index -- No (or multiple) ghc rts package is registered!!
                     -- Feh, whatever, the ghc test suite does some crazy stuff.
 
+-- | Check the consistency of the given package databases.
+checkPackageDBs :: Verbosity -> Compiler -> PackageDBStack
+                -> ProgramConfiguration
+                -> IO [(PackageDB, [String])]
+checkPackageDBs verbosity comp packagedbs conf = do
+  checkPackageDbEnvVar
+  checkPackageDbStack comp packagedbs
+  sequenceA
+    [ do strs <- HcPkg.check (hcPkgInfo conf) verbosity packagedb
+         return (packagedb, strs)
+    | packagedb <- packagedbs ]
```

Comment:
A bit surprised this works, since I thought ghc-pkg needs to know about all the package databases under you to check validity (otherwise some package references could look dangling.)